lib/mongo_client.js

```diff
@@ -119,7 +119,6 @@ const validOptions = require('./operations/connect').validOptions;
  * @param {number} [options.bufferMaxEntries=-1] Sets a cap on how many operations the driver will buffer up before giving up on getting a working connection, default is -1 which is unlimited
  * @param {(ReadPreference|string)} [options.readPreference] The preferred read preference (ReadPreference.PRIMARY, ReadPreference.PRIMARY_PREFERRED, ReadPreference.SECONDARY, ReadPreference.SECONDARY_PREFERRED, ReadPreference.NEAREST)
  * @param {object} [options.pkFactory] A primary key factory object for generation of custom _id keys
- * @param {object} [options.promiseLibrary] A Promise library class the application wishes to use such as Bluebird, must be ES6 compatible
```

Comment:
At very least we should throw some warning when users try to do this, otherwise we might silently break user expectations. I think it's probably fine to still support `promiseLibrary` at the `MongoClient` level, and that option just internally sets the "provided" Promise

Reply:
agreed